### PR TITLE
Object pooling

### LINKED ISSUE #114: Use object pooling to improve performance
https://learn.unity.com/tutorial/introduction-to-object-pooling

### PR DESCRIPTION
**Pull request by Sirkular**
Bullets and Artillery are now pooled for performance. When either is requested, an object pulled from an available pool- if there is none available, a new object is created and added to the pool.

Syntax for requesting any prefab:
ObjectPool.Create(string prefabName)

destroying any prefab:
ObjectPool.Destroy(string prefabName, GameObject gameObject)

To see the effects of this, watch the Hierarchy list while the game is in play. Bullets grey out when "destroyed" as they are set inactive and placed in the available pool, then become white again as they are set active and reused.

closes #114

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,5 @@
+root = true
+
+[*.{cs,vb}]
+indent_style = space
+indent_size = 4
```
Comment (MaXeraph) [inline .editorconfig:5]: I added this in.

**Comment by MaXeraph** [comment]
I noticed we also do `Instantiate(Resources.Load(<>))` in:
- [HealingPill.cs](https://github.com/MaXeraph/Adrenarun/blob/object-pooling/Assets/Scripts/HealingPill.cs#L11)
- [EnemyFactory.cs](https://github.com/MaXeraph/Adrenarun/blob/object-pooling/Assets/Scripts/Enemy/EnemyFactory.cs#L68)

And a couple more UI that we probably don't need to pool like:
- [PlayerCentral.cs](https://github.com/MaXeraph/Adrenarun/blob/object-pooling/Assets/Scripts/Player/PlayerCentral.cs#L113) (probably fine to not have to pool this)
- [CompassUI.cs](https://github.com/MaXeraph/Adrenarun/blob/object-pooling/Assets/Scripts/UI/CompassUI.cs#L42) (probably fine to skip)

**Review by MaXeraph — APPROVED** [review]
Check over some of the files I mentioned but other than that it looks good. Thanks for getting this done

**Comment by Sirkular** [comment]
> Would this entails that there's a limit on some prefabs being "alive" at any point in the game ?

No, as a new object is created if there are none to be allocated.

**Comment by MaXeraph** [comment]
> > Would this entails that there's a limit on some prefabs being "alive" at any point in the game ?
> 
> 
> 
> No, as a new object is created if there are none to be allocated.

So based of how often the objects are destroyed we could optimize it there. So each prefab needs to be deleted at some point and not float infinitely.

**Comment by Sirkular** [comment]
> So based of how often the objects are destroyed we could optimize it there. So each prefab needs to be deleted at some point and not float infinitely.

Yup, we could use a caching algorithm. However, I don't think that it's necessary since the objects that are using the pooled system are used so often.

**Review by zhuandy42 — APPROVED** [review]
lgtm

**Pull request merged**